lib/machine/configure.ts

```diff
@@ -132,7 +154,20 @@ export function configure<T extends SdmContext = PushListenerInvocation>(
                         configuration: cfgToUse,
                     });
 
-                const configured = await configurer(sdm);
+                // Decorate the createGoals method onto the SDM
+                (sdm as any).createGoals = async (creator: GoalCreator<any>, configurers: GoalConfigurer<any> | Array<GoalConfigurer<any>>) => {
+                    const gc = await creator(sdm);
+                    if (!!configurers) {
+                        for (const c of toArray(configurers)) {
+                            await c(sdm, gc);
```

Comment:
try/catch?

Reply:
Adding

Reply:
I was adding this but I'm not sure what to do in the catch block. This is a serious error which ideally I want to propagate all the way up and prevent the SDM from starting.

Simply logging and re-throwing the error seems odd. What do you recommend @ddgenome? 

I'd leave it like it is and let the startup crash but happy to hear your thoughts.

Reply:
I'd either not try/catch or log and re-throw. The latter is a pattern I do often.

```typescript
try {
    // something
} catch (e) {
    e.message = `Add some context: ${e.message}`;
    progressLog.write(e.message);
    logger.error(e.message):
    throw e;
}
```

Reply:
Sounds good. I’ll add the pattern and use it elsewhere.